testkit/tests/runtime.rs

```diff
@@ -0,0 +1,226 @@
+unsafe impl Send for RuntimeTester {}
+unsafe impl Sync for RuntimeTester {}
+
+#[derive(Debug)]
+struct TestRuntime {
+    tester: Arc<RuntimeTester>,
+}
+
+impl TestRuntime {
+    // Runtime identifier.
+    const ID: u32 = 42;
+
+    pub fn with_runtime_tester(tester: Arc<RuntimeTester>) -> Self {
+        TestRuntime { tester }
+    }
+}
+
+impl Runtime for TestRuntime {
+    fn deploy_artifact(
+        &mut self,
+        artifact: ArtifactId,
+        deploy_spec: Any,
+    ) -> Box<dyn Future<Item = (), Error = ExecutionError>> {
+        self.tester.deploy_artifact(artifact, deploy_spec);
+        Box::new(Ok(()).into_future())
+    }
+
+    fn is_artifact_deployed(&self, id: &ArtifactId) -> bool {
+        self.tester.is_artifact_deployed(id)
+    }
+
+    fn artifact_protobuf_spec(&self, _id: &ArtifactId) -> Option<ArtifactProtobufSpec> {
+        Some(ArtifactProtobufSpec::default())
+    }
+
+    fn start_service(&mut self, _spec: &InstanceSpec) -> Result<(), ExecutionError> {
+        Ok(())
+    }
+
+    fn configure_service(
+        &self,
+        _fork: &Fork,
+        _descriptor: InstanceDescriptor,
+        parameters: Any,
+    ) -> Result<(), ExecutionError> {
+        self.tester.configure_service(parameters);
+        Ok(())
+    }
+
+    fn stop_service(&mut self, _descriptor: InstanceDescriptor) -> Result<(), ExecutionError> {
+        Ok(())
+    }
+
+    fn execute(
+        &self,
+        _context: &ExecutionContext,
+        _call_info: &CallInfo,
+        _arguments: &[u8],
+    ) -> Result<(), ExecutionError> {
+        Ok(())
+    }
+
+    fn state_hashes(&self, _snapshot: &dyn Snapshot) -> StateHashAggregator {
+        StateHashAggregator::default()
+    }
+
+    fn before_commit(&self, _dispatcher: &Dispatcher, _fork: &mut Fork) {}
+
+    fn after_commit(
+        &self,
+        _dispatcher: &DispatcherSender,
+        _snapshot: &dyn Snapshot,
+        _service_keypair: &(PublicKey, SecretKey),
+        _tx_sender: &ApiSender,
+    ) {
+    }
+}
+
+impl From<TestRuntime> for (u32, Box<dyn Runtime>) {
+    fn from(inner: TestRuntime) -> Self {
+        (TestRuntime::ID, Box::new(inner))
+    }
+}
+
+#[derive(Debug)]
+struct TestRuntimeFactory {
+    runtime_tester: Arc<RuntimeTester>,
+}
+
+impl TestRuntimeFactory {
+    fn with_runtime_tester(runtime_tester: Arc<RuntimeTester>) -> Self {
+        TestRuntimeFactory { runtime_tester }
+    }
+}
+
+impl RuntimeFactory for TestRuntimeFactory {
+    fn create_runtime(&self) -> (u32, Box<dyn Runtime>) {
+        TestRuntime::with_runtime_tester(self.runtime_tester.clone()).into()
+    }
+}
+
+// We assert that:
+//  1) TestRuntime was passed to the testing blockchain
+//  2) Artifact was deployed with correct deploy specification
+//  3) Service was instantiated with correct initialization parameters
+#[test]
+fn test_runtime_factory() {
+    let tester = Arc::new(RuntimeTester::default());
```

Comment:
Will `Rc` work?

Reply:
No, `Runtime` should be `Send`, `RuntimeFactory` - `Send` + `Sync`